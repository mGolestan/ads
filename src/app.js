// @flow
import express from "express";
import bodyParser from "body-parser";
import compress from "compression";
import routes from "./routes";
import helmet from "helmet";
import cors from "cors";
import loadEnv from "./utils/loadEnv";
import { notFound, errResponse, logRequest } from "./middlewares";

const app = express();

app
  .use(
    helmet({
      dnsPrefetchControl: false,
      frameguard: false,
      noCache: true
    })
  )
  .use(compress())
  .use(cors("*"))
  .use(bodyParser.json())
  .use(bodyParser.urlencoded({ extended: true }))
  .use(logRequest)
  .use("/", routes)
  .use(notFound)
  .use(errResponse);

module.exports = app;
