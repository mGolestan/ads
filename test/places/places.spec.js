// @flow

import request from "supertest";

import app from "../../src/app";

describe("GET /places", () => {
  it("returns a valid result on a valid request", () => {
    return request(app)
      .get("/places")
      .expect(200)
      .then(res => {
        expect(res.body).toMatchSnapshot();
      });
  });
});
