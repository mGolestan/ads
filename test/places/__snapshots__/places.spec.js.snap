// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /places returns a valid result on a valid request 1`] = `
Object {
  "results": Array [
    Object {
      "id": 1,
      "isNew": false,
      "lat": 35.696111,
      "level": "place2",
      "long": 51.423056,
      "name": "تهران",
      "neighbours": "",
      "ordering": 1,
      "parentId": 0,
      "radius": 25000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "tehran",
    },
    Object {
      "id": 2,
      "isNew": false,
      "lat": 35.832677,
      "level": "place2",
      "long": 50.927498,
      "name": "کرج",
      "neighbours": "",
      "ordering": 2,
      "parentId": 0,
      "radius": 25000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "karaj",
    },
    Object {
      "id": 3,
      "isNew": false,
      "lat": 36.301355,
      "level": "place2",
      "long": 59.604209,
      "name": "مشهد",
      "neighbours": "",
      "ordering": 3,
      "parentId": 0,
      "radius": 20000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "mashhad",
    },
    Object {
      "id": 4,
      "isNew": false,
      "lat": 32.633333,
      "level": "place2",
      "long": 51.65,
      "name": "اصفهان",
      "neighbours": "",
      "ordering": 4,
      "parentId": 0,
      "radius": 20000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "isfahan",
    },
    Object {
      "id": 5,
      "isNew": false,
      "lat": 38.066667,
      "level": "place2",
      "long": 46.3,
      "name": "تبریز",
      "neighbours": "",
      "ordering": 5,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "tabriz",
    },
    Object {
      "id": 6,
      "isNew": false,
      "lat": 29.616667,
      "level": "place2",
      "long": 52.533333,
      "name": "شیراز",
      "neighbours": "",
      "ordering": 6,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "shiraz",
    },
    Object {
      "id": 7,
      "isNew": false,
      "lat": 31.328978,
      "level": "place2",
      "long": 48.685473,
      "name": "اهواز",
      "neighbours": "",
      "ordering": 7,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "ahvaz",
    },
    Object {
      "id": 8,
      "isNew": false,
      "lat": 34.65,
      "level": "place2",
      "long": 50.88,
      "name": "قم",
      "neighbours": "",
      "ordering": 8,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "qom",
    },
    Object {
      "id": 9,
      "isNew": false,
      "lat": 34.334081,
      "level": "place2",
      "long": 47.076058,
      "name": "کرمانشاه",
      "neighbours": "",
      "ordering": 9,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "kermanshah",
    },
    Object {
      "id": 10,
      "isNew": false,
      "lat": 37.548312,
      "level": "place2",
      "long": 45.064736,
      "name": "ارومیه",
      "neighbours": "",
      "ordering": 10,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "urmia",
    },
    Object {
      "id": 11,
      "isNew": false,
      "lat": 29.492057,
      "level": "place2",
      "long": 60.847335,
      "name": "زاهدان",
      "neighbours": "",
      "ordering": 11,
      "parentId": 0,
      "radius": 18000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "zahedan",
    },
    Object {
      "id": 12,
      "isNew": false,
      "lat": 37.280609,
      "level": "place2",
      "long": 49.590197,
      "name": "رشت",
      "neighbours": "",
      "ordering": 12,
      "parentId": 0,
      "radius": 20000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "rasht",
    },
    Object {
      "id": 13,
      "isNew": false,
      "lat": 30.281899,
      "level": "place2",
      "long": 57.067394,
      "name": "کرمان",
      "neighbours": "",
      "ordering": 13,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "kerman",
    },
    Object {
      "id": 14,
      "isNew": false,
      "lat": 34.800554,
      "level": "place2",
      "long": 48.516941,
      "name": "همدان",
      "neighbours": "",
      "ordering": 14,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "hamedan",
    },
    Object {
      "id": 15,
      "isNew": false,
      "lat": 34.092473,
      "level": "place2",
      "long": 49.696541,
      "name": "اراک",
      "neighbours": "",
      "ordering": 15,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "arak",
    },
    Object {
      "id": 16,
      "isNew": false,
      "lat": 31.890968,
      "level": "place2",
      "long": 54.356117,
      "name": "یزد",
      "neighbours": "",
      "ordering": 16,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "yazd",
    },
    Object {
      "id": 17,
      "isNew": false,
      "lat": 38.240877,
      "level": "place2",
      "long": 48.296728,
      "name": "اردبیل",
      "neighbours": "",
      "ordering": 17,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "ardabil",
    },
    Object {
      "id": 18,
      "isNew": false,
      "lat": 27.195098,
      "level": "place2",
      "long": 56.308336,
      "name": "بندرعباس",
      "neighbours": "",
      "ordering": 18,
      "parentId": 0,
      "radius": 15000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "bandar-abbas",
    },
    Object {
      "id": 19,
      "isNew": false,
      "lat": 36.267321,
      "level": "place2",
      "long": 50.003886,
      "name": "قزوین",
      "neighbours": "",
      "ordering": 19,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "qazvin",
    },
    Object {
      "id": 20,
      "isNew": false,
      "lat": 36.679433,
      "level": "place2",
      "long": 48.496056,
      "name": "زنجان",
      "neighbours": "",
      "ordering": 20,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "zanjan",
    },
    Object {
      "id": 21,
      "isNew": false,
      "lat": 36.835943,
      "level": "place2",
      "long": 54.441376,
      "name": "گرگان",
      "neighbours": "",
      "ordering": 21,
      "parentId": 0,
      "radius": 8000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "gorgan",
    },
    Object {
      "id": 22,
      "isNew": false,
      "lat": 36.562876,
      "level": "place2",
      "long": 53.060875,
      "name": "ساری",
      "neighbours": "",
      "ordering": 22,
      "parentId": 0,
      "radius": 5000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "sari",
    },
    Object {
      "id": 23,
      "isNew": false,
      "lat": 32.38547,
      "level": "place2",
      "long": 48.400612,
      "name": "دزفول",
      "neighbours": "",
      "ordering": 23,
      "parentId": 0,
      "radius": 8000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "dezful",
    },
    Object {
      "id": 24,
      "isNew": false,
      "lat": 30.347398,
      "level": "place2",
      "long": 48.297272,
      "name": "آبادان",
      "neighbours": "",
      "ordering": 24,
      "parentId": 0,
      "radius": 6000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "abadan",
    },
    Object {
      "id": 25,
      "isNew": false,
      "lat": 28.922232,
      "level": "place2",
      "long": 50.833912,
      "name": "بوشهر",
      "neighbours": "",
      "ordering": 25,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "bushehr",
    },
    Object {
      "id": 26,
      "isNew": false,
      "lat": 33.893502,
      "level": "place2",
      "long": 48.760706,
      "name": "بروجرد",
      "neighbours": "",
      "ordering": 26,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "borujerd",
    },
    Object {
      "id": 27,
      "isNew": false,
      "lat": 33.473836,
      "level": "place2",
      "long": 48.345921,
      "name": "خرم‌آباد",
      "neighbours": "",
      "ordering": 27,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "khorramabad",
    },
    Object {
      "id": 28,
      "isNew": false,
      "lat": 35.298315,
      "level": "place2",
      "long": 47.006275,
      "name": "سنندج",
      "neighbours": "",
      "ordering": 28,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "sanandaj",
    },
    Object {
      "id": 29,
      "isNew": false,
      "lat": 35.541166,
      "level": "place2",
      "long": 51.236012,
      "name": "اسلام‌شهر",
      "neighbours": "",
      "ordering": 29,
      "parentId": 0,
      "radius": 5000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "eslamshahr",
    },
    Object {
      "id": 30,
      "isNew": false,
      "lat": 33.982656,
      "level": "place2",
      "long": 51.424547,
      "name": "کاشان",
      "neighbours": "",
      "ordering": 30,
      "parentId": 0,
      "radius": 5000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "kashan",
    },
    Object {
      "id": 31,
      "isNew": false,
      "lat": 32.63128,
      "level": "place2",
      "long": 51.358921,
      "name": "نجف‌آباد",
      "neighbours": "",
      "ordering": 31,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "najafabad",
    },
    Object {
      "id": 32,
      "isNew": false,
      "lat": 33.631058,
      "level": "place2",
      "long": 46.408971,
      "name": "ایلام",
      "neighbours": "",
      "ordering": 32,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "ilam",
    },
    Object {
      "id": 33,
      "isNew": false,
      "lat": 26.537859,
      "level": "place2",
      "long": 53.975804,
      "name": "کیش",
      "neighbours": "",
      "ordering": 33,
      "parentId": 0,
      "radius": 5000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "kish",
    },
    Object {
      "id": 34,
      "isNew": false,
      "lat": 32.882038,
      "level": "place2",
      "long": 59.21079,
      "name": "بیرجند",
      "neighbours": "",
      "ordering": 34,
      "parentId": 0,
      "radius": 10000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "birjand",
    },
    Object {
      "id": 35,
      "isNew": false,
      "lat": 35.574962,
      "level": "place2",
      "long": 53.395761,
      "name": "سمنان",
      "neighbours": "",
      "ordering": 35,
      "parentId": 0,
      "radius": 5000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "semnan",
    },
    Object {
      "id": 36,
      "isNew": false,
      "lat": 32.328337,
      "level": "place2",
      "long": 50.863214,
      "name": "شهرکرد",
      "neighbours": "",
      "ordering": 36,
      "parentId": 0,
      "radius": 7000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "shahrekord",
    },
    Object {
      "id": 37,
      "isNew": false,
      "lat": 30.560192,
      "level": "place2",
      "long": 49.19761,
      "name": "بندر ماهشهر",
      "neighbours": "",
      "ordering": 37,
      "parentId": 0,
      "radius": 5000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "mahshahr",
    },
    Object {
      "id": 38,
      "isNew": false,
      "lat": 30.675715,
      "level": "place2",
      "long": 51.586201,
      "name": "یاسوج",
      "neighbours": "",
      "ordering": 38,
      "parentId": 0,
      "radius": 7000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "yasuj",
    },
    Object {
      "id": 39,
      "isNew": false,
      "lat": 37.4784,
      "level": "place2",
      "long": 57.322489,
      "name": "بجنورد",
      "neighbours": "",
      "ordering": 39,
      "parentId": 0,
      "radius": 6000,
      "showInFilter": true,
      "showInInput": true,
      "slug": "bojnurd",
    },
  ],
}
`;
